third_party/move/move-model/src/builder/exp_builder.rs

```diff
@@ -3548,6 +3690,34 @@ impl<'env, 'translator, 'module_translator> ExpTranslator<'env, 'translator, 'mo
                 vec![global_access.into_exp()],
             );
         }
+
+        if let Some(entry) = self.parent.parent.fun_table.get(&global_var_sym) {
+            let module_id = entry.module_id;
+            let fun_id = entry.fun_id;
+            let result_type = entry.result_type.clone();
+            let type_params = entry.type_params.clone();
+            let param_types = entry
+                .params
+                .iter()
+                .map(|param| param.get_type())
+                .collect_vec();
+            let Some(instantiation) =
+                self.translate_entry_instantiation(loc, &type_params, type_args)
+            else {
+                return self.new_error_exp();
+            };
+            let fun_type = Type::Fun(
+                Box::new(Type::tuple(param_types)),
+                Box::new(result_type),
+                AbilitySet::DEFINED_FUNCTIONS,
+            );
+            let fun_type = fun_type.instantiate(&instantiation);
+            let fun_type = self.check_type(loc, &fun_type, expected_type, context);
+
+            let id = self.parent.parent.env.new_node(loc.clone(), fun_type);
+            return ExpData::MoveFunctionExp(id, module_id, fun_id);
```

Comment:
This should be `Call(id, Closure(module_id, fund_id, 0x0), vec![])`

Reduce the number of variants in the IR is a key to maintainable and correct code.